subprojects/language-native/src/main/java/org/gradle/language/swift/plugins/SwiftLibraryPlugin.java

```diff
@@ -89,12 +90,18 @@ public void execute(Project project) {
                 boolean staticLibs = library.getLinkage().get().contains(Linkage.STATIC);
 
                 ToolChainSelector.Result<SwiftPlatform> result = toolChainSelector.select(SwiftPlatform.class);
+                SwiftLanguageVersion swiftLanguageVersion = library.getSwiftLanguageVersionSupport().getOrNull();
```

Comment:
We just removed a whole bunch of duplicated code from these plugins. It would be a shame to add duplicated code back. This looks like some convention or rule that applies to all swift components, so could live somewhere that is responsible for applying logic to all swift components, eg the Swift base plugin.